gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/utils/gcp/gcp_secret_manager_client.py

```diff
@@ -0,0 +1,57 @@
+#   Copyright (C) 2023-2024 Credit Mutuel Arkea
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+"""API Client to consume GCP Secret Manager API"""
+
+import logging
+from abc import ABC
+from typing import TypeVar
+
+from google.api_core.exceptions import NotFound
+from google.cloud import secretmanager
+
+from gen_ai_orchestrator.configurations.environment.settings import application_settings
+
+logger = logging.getLogger(__name__)
+T = TypeVar('T')
+
+
+class GCPSecretManagerClient(ABC):
```

Comment:
Sorry I was not thinking of using ABC here but more like this : 
```python
import abc

class AbstractSecretManager(metaclass=abc.ABCMeta):

  @abc.abstractmethod
  def get_secret(self, secret_id: str) -> str:
      return

class GCPSecretManagerClient(AbstractSecretManager):
.....

class AWSSecretManagerClient(AbstractSecretManager):
.....
```

Reply:
Yes, I had done it on the aws side, but not on the gcp implementation side.